src/MassTransit/Configuration/MessageDataConfiguratorExtensions.cs

```diff
@@ -21,5 +24,32 @@ public static void UseMessageData(this IBusFactoryConfigurator configurator, IMe
                 var observer = new CourierMessageDataConfigurationObserver(configurator, repository, false);
             }
         }
+
+        /// <summary>
+        /// Enable the loading of message data for the any message type that includes a MessageData property.
+        /// </summary>
+        /// <param name="configurator">The bus factory configurator.</param>
+        /// <param name="selector">The repository selector. See extension methods, e.g. <see cref="MessageDataRepositorySelectorExtensions.WithFileSystemStorage"/>.</param>
+        public static void UseMessageData(this IBusFactoryConfigurator configurator, Func<IMessageDataRepositorySelector, IMessageDataRepository> selector)
+        {
+            if (configurator is null)
+            {
+                throw new ArgumentNullException(nameof(configurator));
+            }
+
+            if (selector is null)
+            {
+                throw new ArgumentNullException(nameof(selector));
+            }
+
+            var repository = selector(new MessageDataRepositorySelector(configurator));
+
+            UseMessageData(configurator, repository);
+
+            if (repository is IBusObserver observer)

```

Comment:
Ideally this should reside in `UseMessageData(this IBusFactoryConfigurator configurator, IMessageDataRepository repository)` but that could unexpected behavior I imagine if existing consumers manually observe the bus.